### PR TITLE
Add 'oc rsh' for simpler access to a remote shell

### PR DESCRIPTION
**Pull request by smarterclayton**
@gshipley as you wish

### CODE REVIEW

**Comment by liggitt** [comment]
Is `rsh` going to imply that it's actually using remote shell (even though it's not), and will that scare people who know remote shell is largely insecure (https://en.wikipedia.org/wiki/Remote_Shell#Limitations)?

File: pkg/cmd/cli/cmd/rsh.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	rshLong = `
+Open a remote shell session to a container
+
+This command will attempt to start a shell session in the specified pod. It will default to the
+first container if none is specified, and will attempt to use '/bin/bash' as the default shell.
+Note, some containers may not include a shell - use '%[1]s exec' if you need to run commands
+directly.`
+
+	rshExample = `
+  // Open a shell session on the first container in pod 'foo'
```
Comment (kargakis) [inline pkg/cmd/cli/cmd/rsh.go]: Why not provide `-c` and specify the container as well?

**Comment by smarterclayton** [comment]
I don't want to call it ssh and I think we might want to reserve sh
possibly.  Open to suggestions or other opinions.  Also, help text should
say "secure".

It'll be paired with rsync and/or cp.  maybe other helpers.  Ideas?

On Jul 23, 2015, at 1:08 AM, Jordan Liggitt <notifications@github.com>
wrote:

Is rsh going to imply that it's actually using remote shell (even though
it's not), and will that scare people who know remote shell is largely
insecure (https://en.wikipedia.org/wiki/Remote_Shell#Limitations)?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3863#issuecomment-123965958>.

File: pkg/cmd/cli/cmd/rsh.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/rsh.go]: We do?

File: pkg/cmd/cli/cmd/rsh.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		In:  in,
+		Out: out,
+		Err: err,
+
+		TTY:   true,
+		Stdin: true,
+
+		Executor: &kubecmd.DefaultRemoteExecutor{},
+	}
+	executable := "/bin/bash"
```
Comment (soltysh) [inline pkg/cmd/cli/cmd/rsh.go]: What if there's no `/bin/bash`? Eg. busybox based images have only `/bin/sh`.

Comment (soltysh) [inline pkg/cmd/cli/cmd/rsh.go]: Nvmd, now I noticed an option to switch it.

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**